tlv/fuzz_test.go

```diff
@@ -168,3 +168,111 @@ func FuzzTUint64(f *testing.F) {
 		harness(t, data, ETUint64, DTUint64, &val, uint64(decodeLen))
 	})
 }
+
+// encodeParsedTypes re-encodes TLVs decoded from a stream, using the
+// parsedTypes and decodedRecords produced during decoding.
+func encodeParsedTypes(t *testing.T, parsedTypes TypeMap,
+	decodedRecords []Record) []byte {
+
+	var encodeRecords []Record
+	for typ, val := range parsedTypes {
+		if typ < Type(len(decodedRecords)) {
+			encodeRecords = append(
+				encodeRecords, decodedRecords[typ],
+			)
+			continue
+		}
+
+		val := val
+		encodeRecords = append(
+			encodeRecords, MakePrimitiveRecord(typ, &val),
+		)
+	}
+	SortRecords(encodeRecords)
+	encodeStream := MustNewStream(encodeRecords...)
+
+	var b bytes.Buffer
+	require.NoError(t, encodeStream.Encode(&b))
+
+	return b.Bytes()
+}
+
+// FuzzStream does two stream decode-encode cycles on the fuzzer data and checks
+// that the encoded values match.
+func FuzzStream(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var (
+			u8   uint8
+			u16  uint16
+			u32  uint32
+			u64  uint64
+			b32  [32]byte
+			b33  [33]byte
+			b64  [64]byte
+			pk   *btcec.PublicKey
+			b    []byte
+			bs32 uint32
```

Comment:
curious about what `bs` and `tu` stand for?

Reply:
BigSize and Truncated Uint.